src/AutoRest.CSharp/Common/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -47,8 +47,8 @@ public async Task<GeneratedCodeWorkspace> ExecuteAsync(CodeModel codeModel)
                 }
                 else
                 {
-                    MgmtContext.Initialize(new BuildContext<MgmtOutputLibrary>(codeModel, sourceInputModel, schemaUsageProvider));
                     CodeModelTransformer.TransformForMgmt(codeModel);
+                    MgmtContext.Initialize(new BuildContext<MgmtOutputLibrary>(codeModel, sourceInputModel, schemaUsageProvider));
```

Comment:
I think before we go any further, we should remove this `MgmtContext` first to make everything explicit.
Also we should remove `BuildContext`.